### PR TITLE
Issue 1489: Reduce email sends to first-time transition for the applicable transitions

### LINKED ISSUE #1489: Reduce email notifications for some status changes
### Issue description

As a domain requestor
I want to receive notifications of status changes ONLY when those changes initially occur
So that I do not get excessive emails which are not meaningful to me

Existing behavior: A notification email is sent to the requestor upon each occurrence of a status change. (Submitted, withdrawn, approved, rejected)

### Acceptance criteria

- [x] No notification is sent when the request moves from "Action Needed" to "Submitted" (requester re-submits with changes)
- [x] No notification is sent when the request moves from "In Review" back to "Submitted" (analyst returned it to the queue)
- [x] #1582

### Additional context

Consider the following statuses, listed horizontally, with the possible transitions listed vertically below each.
Image

For those statuses that have the corresponding email icon, a notification email should be sent when the request INITIALLY enters that status. Subsequent status changes may take place which might return the request to a previously held status. When this happens, a second email should NOT be sent to the requestor as this adds no valuable information and may even be confusing to the user.
 
For example, a request may be `submitted` by the requestor, which triggers a notification email confirming the submission. Later, the request may be moved to `In Review` (which does not use a notification). If this request is moved back to `submitted` status, this does NOT trigger another notification email, as it has already been in that status.

### Links to other issues

Related to:  #1582 

_No response_

### PR DESCRIPTION
**Pull request by rachidatecs**
## Ticket

Resolves #1489 

## Changes

- Skip email sends when moving from in review or action needed to submitted

## Context for reviewers

The model tests can be streamlined. I might revisit tomorrow.

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [x] Created/modified automated tests
- [x] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [x] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [x] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [x] Interactions with external systems are wrapped in try/except
- [x] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

**Review by neilmb — APPROVED** [review]
I really like all the test cases, but I would like them to be a little less repetitive.

File: src/registrar/tests/test_models.py [hunk 1]
```diff
@@ -161,33 +161,158 @@ def test_status_fsm_submit_succeed(self):
...
+                [
+                    email
+                    for email in MockSESClient.EMAILS_SENT
+                    if "mayor@igorville.gov" in email["kwargs"]["Destination"]["ToAddresses"]
+                ]
+            ),
+            0,
+        )
+
+    def test_reject_sends_email(self):
```
[the remark this answers is not in the document]
Comment (rachidatecs) [reply src/registrar/tests/test_models.py:295]: Yeaaaah I got lazy with the models tests. Way to call me out on it, Neil.

[the remark this answers is not in the document]
Comment (rachidatecs) [reply src/registrar/tests/test_models.py:295]: I'll refactor.

**Comment by zandercymatics** [comment]
Assigning "Neil" as the reviewer so he gets credit for the review

File: src/registrar/models/domain_application.py [hunk 2]
```diff
@@ -654,11 +654,13 @@ def submit(self):
         self.submission_date = timezone.now().date()
         self.save()
 
-        self._send_status_update_email(
-            "submission confirmation",
-            "emails/submission_confirmation.txt",
-            "emails/submission_confirmation_subject.txt",
-        )
+        # Limit email notifications to transitions from Started and Withdrawn
+        if self.status == self.ApplicationStatus.STARTED or self.status == self.ApplicationStatus.WITHDRAWN:
```
Comment (zandercymatics) [inline src/registrar/models/domain_application.py:658]:
```suggestion
        limited_statuses = [self.ApplicationStatus.STARTED, self.ApplicationStatus.WITHDRAWN]
        # Limit email notifications to transitions from Started and Withdrawn
        if self.status in limited_statuses:
```
(Optional) While I doubt it, its possible that this list may be expanded in the future so storing it in an array like this makes it easier to expand

File: src/registrar/tests/test_admin.py [hunk 3]
```diff
@@ -411,29 +412,27 @@ def test_short_org_name_in_applications_list(self):
         # Now let's make sure the long description does not exist
         self.assertNotContains(response, "Federal: an agency of the U.S. government")
 
-    @boto3_mocking.patching
-    def test_save_model_sends_submitted_email(self):
-        # make sure there is no user with this email
-        EMAIL = "mayor@igorville.gov"
-        User.objects.filter(email=EMAIL).delete()
+    def transition_state_and_send_email(self, application, status):
+        """Helper method for the email test cases."""
```
Comment (zandercymatics) [inline src/registrar/tests/test_admin.py:416]: Nice

**Review by Katherine-Osos — APPROVED** [review]

**Pull request merged**